diff -r -u llvm-3.7.1.src/lib/Support/regcomp.c /opt/llvm/llvm-3.7.1.src/lib/Support/regcomp.c
--- a/lib/Support/regcomp.c	2015-02-26 21:07:30.000000000 +0000
+++ b/lib/Support/regcomp.c	2017-01-28 00:20:46.815680933 +0000
@@ -1085,12 +1085,12 @@
 		ptr = (cset *)realloc((char *)p->g->sets, nc * sizeof(cset));
 		if (ptr == NULL)
 			goto nomem;
-		p->g->sets = ptr;
+		p->g->sets = (cset*)ptr;
 
 		ptr = (uch *)realloc((char *)p->g->setbits, nbytes);
 		if (ptr == NULL)
 			goto nomem;
-		p->g->setbits = ptr;
+		p->g->setbits = (uch*)ptr;
 
 		for (i = 0; i < no; i++)
 			p->g->sets[i].ptr = p->g->setbits + css*(i/CHAR_BIT);
@@ -1223,7 +1223,7 @@
 		SETERROR(REG_ESPACE);
 		return;
 	}
-	cs->multis = np;
+	cs->multis = (char*)np;
 
 	llvm_strlcpy(cs->multis + oldend - 1, cp, cs->smultis - oldend + 1);
 }
@@ -1523,7 +1523,7 @@
 		return;
 
 	/* turn it into a character string */
-	g->must = malloc((size_t)g->mlen + 1);
+	g->must = (char*)malloc((size_t)g->mlen + 1);
 	if (g->must == NULL) {		/* argh; just forget it */
 		g->mlen = 0;
 		return;
diff -r -u llvm-3.7.1.src/lib/Support/regexec.c /opt/llvm/llvm-3.7.1.src/lib/Support/regexec.c
--- a/lib/Support/regexec.c	2012-08-24 23:29:28.000000000 +0000
+++ b/lib/Support/regexec.c	2017-01-28 00:21:06.511272479 +0000
@@ -111,7 +111,7 @@
 #define	ASSIGN(d, s)	memmove(d, s, m->g->nstates)
 #define	EQ(a, b)	(memcmp(a, b, m->g->nstates) == 0)
 #define	STATEVARS	long vn; char *space
-#define	STATESETUP(m, nv)	{ (m)->space = malloc((nv)*(m)->g->nstates); \
+#define	STATESETUP(m, nv)	{ (m)->space = (char*)malloc((nv)*(m)->g->nstates); \
 				if ((m)->space == NULL) return(REG_ESPACE); \
 				(m)->vn = 0; }
 #define	STATETEARDOWN(m)	{ free((m)->space); }
